src/test/java/com/google/devtools/build/lib/rules/cpp/Cpp20ModulesConfiguredTargetTest.java

```diff
@@ -0,0 +1,111 @@
+package com.google.devtools.build.lib.rules.cpp;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.devtools.build.lib.analysis.ConfiguredRuleClassProvider;
+import com.google.devtools.build.lib.analysis.util.AnalysisMock;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.analysis.util.DummyTestFragment;
+import com.google.devtools.build.lib.packages.util.Crosstool;
+import com.google.devtools.build.lib.testutil.TestRuleClassProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class Cpp20ModulesConfiguredTargetTest extends BuildViewTestCase {
+
+  @Override
+  protected ConfiguredRuleClassProvider createRuleClassProvider() {
+    ConfiguredRuleClassProvider.Builder builder = new ConfiguredRuleClassProvider.Builder();
+    TestRuleClassProvider.addStandardRules(builder);
+    builder.addConfigurationFragment(DummyTestFragment.class);
+    return builder.addRuleDefinition(new TestRuleClassProvider.MakeVariableTesterRule()).build();
+  }
+
+  @Test
+  public void testCpp20ModulesConfigurationNoFlags() throws Exception {
+    ImmutableList<String> targetList = ImmutableList.of("//foo:lib", "//foo:bin", "//foo:test");
+    scratch.file(
+        "foo/BUILD",
+        """
+                cc_library(
+                    name = 'lib',
+                    module_interfaces = ["foo.cppm"],
+                )
+                cc_binary(
+                    name = 'bin',
+                    module_interfaces = ["foo.cppm"],
+                )
+                cc_test(
+                    name = 'test',
+                    module_interfaces = ["foo.cppm"],
+                )
+                """);
+    for(String targetName: targetList) {
```

Comment:
Unroll the loop, i.e copy-paste each of the assert without using a `for`. Same for the other two tests.

This is where you don't want to follow DRY in the test. Without a loop when an assert fails, the line number will tell you where it failed. (With a for loop you can't tell which `targetName` failed)